src/client/lobby/room-creator.js

```diff
@@ -4,34 +4,39 @@ import config from '../config/config';
 
 import style from './lobby.scss';
 
-export default class RoomCreator extends Component {
-
-    onUsernameClick(){
-        var username = document.getElementById('user-name').value;
-        this.props.setUsername(username);
-    }
+var username;
+var roomname;
 
-    onRoomnameClick(){
-        var roomname = document.getElementById('room-name').value;
-        this.props.setRoomname(roomname);
+export default class RoomCreator extends Component {
+    componentDidMount(){
+        username = document.getElementById('user-name');
+        roomname = document.getElementById('room-name');
     }
-
     render() {
         return (
             <div className="col-lg-4 text-center">
                 <h2>Create Room & Set Username</h2>
                 <div className="col-lg-12 list-group">
                     <div className={`row ${style.space}`}>
                         <input id="user-name" type="text" placeholder="Please type your new username"></input>
-                        <button id="user-name-button" className="btn btn-success" onClick={this.onUsernameClick.bind(this)}>Set Username</button>
+                        <button id="user-name-button" className="btn btn-success" onClick={onUsernameClick.bind(this)}>Set Username</button>
```

Comment:
Please change to `onClick={() => onUsernameClick()}`. It's nicer than the bind syntax :smile_cat: